R/file.R

```diff
@@ -27,8 +27,12 @@ read_file <- function(file, locale = default_locale()) {
   if (empty_file(file)) {
     return("")
   }
-
-  ds <- datasource(file)
+  if (missing(locale)) {
```

Comment:
This code is repeated in a lot of places. Could you pull out into a separate function?  

I'm also not sure why you have different logic when the locale is missing.